lib/charms/postgresql_k8s/v0/postgresql.py

```diff
@@ -165,15 +165,20 @@ def create_user(
                 user_definition = (
                     f"WITH LOGIN{' SUPERUSER' if admin else ''} ENCRYPTED PASSWORD '{password}'"
                 )
-                if extra_user_roles:
-                    user_definition += f' {extra_user_roles.replace(",", " ")}'
+                #if extra_user_roles:
+                #    user_definition += f' {extra_user_roles.replace(",", " ")}'
                 if cursor.fetchone() is not None:
                     statement = "ALTER ROLE {}"
                 else:
                     statement = "CREATE ROLE {}"
                 cursor.execute(
                     sql.SQL(f"{statement} {user_definition};").format(sql.Identifier(user))
                 )
+                if extra_user_roles:
```

Comment:
@marceloneppel this was buggy, this fix seems to work, but feel free to solve in other way

Reply:
I saw that this change broke the situations when the client app request privileges like `CREATEDB`, so I improved the library code on https://github.com/canonical/postgresql-operator/pull/137/commits/e8a5bf187f197e3b181e5fc36dcd426d6380d892 to handle correctly both privileges (like `CREATEDB`) and roles (like `pg_monitor`).

Reply:
I also copied the code to the k8s charm (which owns the library) through https://github.com/canonical/postgresql-k8s-operator/pull/158.